homeassistant/components/automation/state.py

```diff
@@ -101,6 +110,11 @@ def if_action(hass, config):
     entity_id = config.get(CONF_ENTITY_ID)
     state = config.get(CONF_STATE)
 
+    if CONF_FOR in config:
```

Comment:
Better to rewrite this so that `time_delta` is always available. That way the implementation detail of `CONF_FOR in config` only has to be added here.

```python
time_delta = get_time_config(config)
if CONF_FOR in config and time_delta is None:
    return None
```